File: FrontendClubManagement/src/services/adminDepartmentService.ts
import { axiosClient, type ApiResponse } from "@/api/axiosClient";

export interface CampusSimpleResponse {
  id: number;
  campusCode: string;
  campusName: string;
}

export interface AdminDepartmentResponse {
  id: number;
  departmentCode: string;
  departmentName: string;
  email: string;
  phone: string;
  avatarUrl: string;
  bannerUrl: string;
  fbLink: string;
  igLink: string;
  ttLink: string;
  ytLink: string;
  sortDescription: string;
  campus: CampusSimpleResponse;
}

export interface AdminDepartmentUpdateRequest {
  departmentName?: string;
  email?: string;
  phone?: string;
  avatarUrl?: string;
  bannerUrl?: string;
  fbLink?: string;
  igLink?: string;
  ttLink?: string;
  ytLink?: string;
  sortDescription?: string;
}

export const adminDepartmentService = {
  async getDepartmentById(
    id: number
  ): Promise<ApiResponse<AdminDepartmentResponse>> {
    return axiosClient.get<AdminDepartmentResponse>(
      `/admin-departments/${id}`
    );
  },

  async getDepartmentsByCampus(
    campusId?: number
  ): Promise<ApiResponse<AdminDepartmentResponse[]>> {
    const url = campusId 
      ? `/admin-departments?campusId=${campusId}`
      : `/admin-departments`;
    return axiosClient.get<AdminDepartmentResponse[]>(url);
  },

  async updateDepartment(
    id: number,
    data: AdminDepartmentUpdateRequest
  ): Promise<ApiResponse<AdminDepartmentResponse>> {
    return axiosClient.put<AdminDepartmentResponse>(
      `/admin-departments/${id}`,
      data
    );
  },
};

export default adminDepartmentService;

File: BackendClubManagement/src/main/java/com/sep490/backendclubmanagement/service/AdminDepartmentService.java
package com.sep490.backendclubmanagement.service;


import com.sep490.backendclubmanagement.dto.request.AdminDepartmentUpdateRequest;
import com.sep490.backendclubmanagement.dto.response.AdminDepartmentResponse;
import com.sep490.backendclubmanagement.exception.AppException;

import java.util.List;

public interface AdminDepartmentService {

    AdminDepartmentResponse getDepartmentById(Long id) throws AppException;

    List<AdminDepartmentResponse> getDepartmentsByCampus(Long campusId);

    AdminDepartmentResponse updateDepartment(Long id, AdminDepartmentUpdateRequest request) throws AppException;
}
